.gitlab-ci.yml

```diff
@@ -1,37 +1,592 @@
+    - (cd go/src/github.com/mendersoftware/tenantadm && echo TENANTADM_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - git clone https://github.com/mem/oe-meta-go.git || true
+    - (cd oe-meta-go && git checkout -f master)
+    - git clone git://git.openembedded.org/meta-openembedded.git || true
+    - (cd meta-openembedded && git fetch -u -f origin ${META_OPENEMBEDDED_REV}:pr && git checkout pr)
+    - (cd meta-openembedded && echo META_OPENEMBEDDED_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - git clone git://github.com/agherzan/meta-raspberrypi.git || true
+    - (cd meta-raspberrypi && git fetch -u -f origin ${META_RASPBERRYPI_REV}:pr && git checkout pr)
+    - (cd meta-raspberrypi && echo META_RASPBERRYPI_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - git clone git://github.com/mendersoftware/mender-conductor.git go/src/github.com/mendersoftware/mender-conductor || true
+    - (cd go/src/github.com/mendersoftware/mender-conductor && git fetch -u -f origin ${MENDER_CONDUCTOR_REV}:pr && git checkout pr)
+    - (cd go/src/github.com/mendersoftware/mender-conductor && echo MENDER_CONDUCTOR_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - git clone git@github.com:mendersoftware/mender-conductor-enterprise.git go/src/github.com/mendersoftware/mender-conductor-enterprise || true
+    - (cd go/src/github.com/mendersoftware/mender-conductor-enterprise && git fetch -u -f origin ${MENDER_CONDUCTOR_ENTERPRISE_REV}:pr && git checkout pr)
+    - (cd go/src/github.com/mendersoftware/mender-conductor-enterprise && echo MENDER_CONDUCTOR_ENTERPRISE_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - git clone https://github.com/mendersoftware/mender-cli.git go/src/github.com/mendersoftware/mender-cli || true
+    - (cd go/src/github.com/mendersoftware/mender-cli && git fetch -u -f origin ${MENDER_CLI_REV}:pr && git checkout pr)
+    - (cd go/src/github.com/mendersoftware/mender-cli && echo MENDER_CLI_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - git clone https://github.com/mendersoftware/mender-image-tests || true
+    - (cd mender-image-tests && git fetch -u -f origin ${MENDER_IMAGE_TESTS_REV}:pr && git checkout pr)
+    - (cd mender-image-tests && echo MENDER_IMAGE_TESTS_REV_GIT_SHA=$(git rev-parse HEAD) >> ${CI_PROJECT_DIR}/build_revisions.env)
+    - cat ${CI_PROJECT_DIR}/build_revisions.env
+    - tar -czf /tmp/workspace.tar.gz .
+    - mv /tmp/workspace.tar.gz ${CI_PROJECT_DIR}/workspace.tar.gz
+  artifacts:
+    expire_in: 2w
+    paths:
+      - workspace.tar.gz
+      - build_revisions.env
+
+.template_build_test_acc: &build_and_test_acceptance
+  image: teracy/ubuntu:16.04-dind-latest
+  services:
+  - docker:dind
+  tags:
+    - mender-qa-slave
+  before_script:
```

Comment:
we should really put this into a base image as well as the https://github.com/mendersoftware/mender-test-containers - at least all the install only steps

Reply:
Thanks. I'll create a JIRA task in the GitLab optimizations bucket. I have few of these that we can go through in the next QA meeting